docs/deployment.md

```diff
@@ -0,0 +1,248 @@
+        "ta_roles": ["assistant"]
+      }
+    }
+    
+    
+## Enabling Group Project XBlock v2 in a course
+
+To enable the use of Group Project XBlock v2 in the course:
+
+1. Open the course in Studio.
+2. In the top menu choose Settings -> Advances Settings
+3. Add `gp-v2-project` to "Advanced Module List"
+4. Optionally, add `ooyala-player` to "Advanced Module List" - this player is used for video resources
+
+As a result, `Advanced Module List` option should look like that (provided there are no other advanced modules are 
+enabled for course): 
+![Advanced Module List Image](/docs/images/advanced_module_list.png)
+
+
+## (Optional) Notifications integration
+
+Group Project XBlock v2 sends a number of notifications via edx-notifications app. In order to make those notifications
+appear correctly, make sure `NOTIFICATION_CLICK_LINK_URL_MAPS` Django setting contains the following record:
+
+    'open-edx.xblock.group-project-v2.*': '/courses/{course_id}/group_work?activate_block_id={location}'
+    
+If you're using edx-solutions/edx-platform fork, appropriate value is already set. Otherwise, note that 
+`NOTIFICATION_CLICK_LINK_URL_MAPS` are read from `lms.env.json`, so they should be modified there.    
+
+Example:
+
+```
+NOTIFICATION_CLICK_LINK_URL_MAPS = {
+    'open-edx.studio.announcements.*': '/courses/{course_id}/announcements',
+    'open-edx.lms.leaderboard.*': '/courses/{course_id}/cohort',
+    'open-edx.lms.discussions.*': '/courses/{course_id}/discussion/{commentable_id}/threads/{thread_id}',
+    'open-edx.xblock.group-project.*': '/courses/{course_id}/group_work?seqid={activity_location}',
+    'open-edx.xblock.group-project-v2.*': '/courses/{course_id}/group_work?activate_block_id={location}',
+}
+```  
+
+# External/API dependencies of GWv2 
+
+Below is a list of a non-standard dependencies of student-facing functionalities of GWv2. 
+
+## XBlock services
+
+* `settings` GWv2 extensively uses settings defined in `XBLOCK_SETTINGS` in edx lms configuration. It uses
+  `group_project_v2` namespace.  
+* `notifications` Used to send notifications to students events like:   
+    * Project stage submissions opened. 
+    * Project stage submissions due.
+    * File has been uploaded 
+    * Grades have been posted.   
+    Please see `StageNotificationsMixin` that contains all our usages of notifications API. 
+* `courseware_parent_info` Still listed as a required plugin for `GroupActivityXBlock`. 
+   
+  Note: Most probably can be safely removed, it was introduced in the first version of Group Work XBlock
+  and all references to this [service have been removed][removal-of-courseware_parent_info] 
+  
+[removal-of-courseware_parent_info]: https://github.com/open-craft/xblock-group-project-v2/commit/5f00966d11dec9942ffca73e606c81ce58d23917
+
+## Dependencies in XBlock runtime
+
+* `get_real_user` --- used to get the `id` of the user, which is then used to query API, for 
```

Comment:
From what I read in the google groups, it's not totally standardized either. It's there, everybody uses it, but it's treated as first class citezen.